accounts-db/src/accounts_db.rs

```diff
@@ -5074,7 +5074,7 @@ impl AccountsDb {
         &self,
         ancestors: &Ancestors,
         account: &Pubkey,
-        owners: &[&Pubkey],
+        owners: &[Pubkey],
     ) -> Result<usize, MatchAccountOwnerError> {
```

Comment:
Here is the caller; looks like it's done so we can avoid the copy and use `&Pubkey` instead of `Pubkey`
https://github.com/solana-labs/solana/blob/fe598a92735c3c1545f9c20da803b08430beeb73/accounts-db/src/accounts.rs#L646-L681

Reply:
We copy it into the entry at the end anyway, but I think we use the `&Pubkey` to avoid doing 2 copies.

I was thinking we could return a `Result<&Pubkey>`. As it is, we're doing extra bounds checks with the `get` call when we already know it's in the bounds since we just found it.
Does that make sense?